Feature/improve automated backup

schroda opened the pull request.

schroda committed:
Add option to disable cleanup of backups

schroda committed:
Ensure the minimum TTL of backups to 1 day

schroda committed:
Schedule the automated backup on a specific time of the day

Syer10 commented on server/build.gradle.kts:
```diff
@@ -70,6 +70,8 @@ dependencies {
 
     testImplementation(libs.mockk)
+
+    implementation("it.sauronsoftware.cron4j:cron4j:2.2.5")
```
Add it to gradle/libs.gradle.toml

schroda committed:
Introduce scheduler that takes system hibernation time into account

In case the system was hibernating/suspended scheduled task that should have been executed during that time would not get triggered and thus, miss an execution.

To prevent this, this new scheduler periodically checks if the system was suspended and in case it was, triggers any task that missed its last execution

schroda committed:
Use new scheduler
